src/prefect/workers/base.py

```diff
@@ -966,9 +967,17 @@ async def _get_configuration(
     ) -> BaseJobConfiguration:
         deployment = await self._client.read_deployment(flow_run.deployment_id)
         flow = await self._client.read_flow(flow_run.flow_id)
+
+        deployment_vars = deployment.infra_overrides or {}
+        if PREFECT_EXPERIMENTAL_ENABLE_FLOW_RUN_INFRA_OVERRIDES:
```

Comment:
I noticed there's an alternate form for doing this in some places:

```python
if experiment_enabled("flow_run_infra_overrides"): ...
```

Let me know if there's a reason to prefer this check instead of using the setting directly.